docs/graphql.md

```diff
@@ -0,0 +1,144 @@
+# Evergreen GraphQL API
+
+Welcome to the beta version of the Evergreen GraphQL API! This API provides read
+and write access to various pieces of data found in Evergreen. You can use the
+GraphQL Playground, which can be found at https://evergreen.mongodb.com/graphql,
+to experiment with and explore the available data.
+
+To use the API in your own code, you can use the endpoint
+https://evergreen.mongodb.com/graphql/query, or alternatively, leverage code
+generation to generate typings in your preferred language.
+
+## Authentication
+
+To access any GraphQL endpoint, ensure you pass the mci-token cookie along with
+your request to authenticate your session with the Evergreen API, providing read
+and/or write access as required.
+
+## Directives
+
+Evergreen's GraphQL API includes directives that manage permissions for specific
+pieces of data. These directives add additional functionality to fields and
+types. A common directive used in Evergreen's GraphQL API is
+`@requireProjectAccess`. Before calling a query that has this directive, ensure
+the caller has permissions to view a project. You can find the other directives
+[here](https://github.com/evergreen-ci/evergreen/blob/d96942bcf0c26b158b8b1313bd27786f7a7c31a7/graphql/schema/directives.graphql).
+
+## Documentation
+
+The Evergreen GraphQL API is self-documenting, meaning that you can use the
+GraphQL Playground to explore the available data and fields. More comprehensive
+documentation is currently being created to help fully utilize the API.
+
+## Examples
+
+GraphQL queries offer efficient ways to retrieve information in single requests,
+compared to its REST counterpart. Below are a couple of examples:
+
+This query retrieves the latest 5 mainline commits and filters the tasks to only
+include those with a successful `e2e_test` task. Additionally, it fetches the
+log links for each test associated with these tasks. Previously, achieving this
+functionality would require multiple synchronous API calls and application-level
+logic utilizing the traditional REST API. However, with our GraphQL API, it can
+be accomplished in a single declarative request.
+
+```graphql
+{
+  mainlineCommits(
+    options: {projectIdentifier: "spruce", limit: 5}
+    buildVariantOptions: {tasks: "e2e_test", statuses: "success"}
+  ) {
+    versions {
+      version {
+        id
+        buildVariants(options: {tasks: "e2e_test"}) {
```

Comment:
Maybe nit, but why do we have to specify `{tasks: "e2e_test"}` twice?

Reply:
So this can actually just be a variable that is reused. But I didn't know how to express that in a markdown comment, I'll update it!